packages/core/src/crypto/x509/X509Service.ts

```diff
@@ -0,0 +1,116 @@
+import * as x509 from '@peculiar/x509'
+import { injectable } from 'tsyringe'
+
+import { AgentContext } from '../../agent'
+import { Key } from '../Key'
+import { CredoWebCrypto } from '../webcrypto'
+
+import { ExtensionInput, X509Certificate } from './X509Certificate'
+import { X509Error } from './X509Error'
+
+@injectable()
+export class X509Service {
+  /**
+   *
+   * Validate a chain of X.509 certificates according to RFC 5280
+   *
+   * This function requires a list of base64 encoded certificates and, optionally, a certificate that should be found in the chain.
+   * If no certificate is provided, it will just assume the leaf certificate
+   *
+   * The leaf certificate should be the 0th index and the root the last
+   *
+   * Note:
+   *   - Does not check whether a root or intermediate certificate is trusted
+   *
+   * Additional validation:
+   *   - Make sure atleast a single certificate is in the chain
+   */
+  public static async validateCertificateChain(
+    agentContext: AgentContext,
+    {
+      certificateChain,
+      certificate = certificateChain[0],
+      date = new Date(),
+    }: {
+      certificateChain: Array<string>
+      certificate?: string
+      date?: Date
+    }
+  ) {
+    if (!certificate) throw new X509Error('Certificate chain is empty')
```

Comment:
if user passes an empty `cerificateChain`, but also a string to `certificate` this will not throw, even if the `certificateChain` is empty